src/main/java/de/teamlapen/vampirism/misc/SettingsProvider.java

```diff
@@ -0,0 +1,143 @@
+package de.teamlapen.vampirism.misc;
+
+import com.google.common.io.ByteStreams;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import de.teamlapen.lib.lib.util.ResourceLocationTypeAdapter;
+import de.teamlapen.vampirism.VampirismMod;
+import de.teamlapen.vampirism.api.settings.ISettingsProvider;
+import de.teamlapen.vampirism.api.settings.Supporter;
+import net.minecraft.resources.ResourceLocation;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class SettingsProvider implements ISettingsProvider {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final Gson GSON = new GsonBuilder().registerTypeHierarchyAdapter(ResourceLocation.class, new ResourceLocationTypeAdapter()).create();
+
+    private final HttpClient client;
+    private final String baseUrl;
+    private final String apiVersion;
+    private final Map<String, String> settingValues = new HashMap<>();
+
+    public SettingsProvider(String baseUrl, String apiVersion) {
+        this.baseUrl = baseUrl;
+        this.apiVersion = apiVersion;
+        this.client = HttpClient.newHttpClient();
+    }
+
+    public void syncSettingsCache() {
+        getSettingValuesAsync().handleAsync(this::checkSettings).thenAccept(x -> {
+            this.settingValues.clear();
+            this.settingValues.putAll(x);
+        });
+    }
+
+    @Nullable
+    public String getSettingsValue(String key) {
+        return this.settingValues.get(key);
+    }
+
+    public @NotNull Optional<String> getSettingsValueOpt(String key) {
+        return Optional.ofNullable(this.settingValues.get(key));
+    }
+
+    public boolean isSettingTrue(String key) {
```

Comment:
Maybe it would be better to have a "default" parameter. Or just get rid of the nullable methods completely and only have optionals. (Optional<String> and Optional<Boolean> and Optional<Integer>)